internal/auth0/organization/data_source.go

```diff
@@ -0,0 +1,237 @@
+					Type:     schema.TypeBool,
+					Computed: true,
+					Description: "When `true`, all users that log in with this connection will be " +
+						"automatically granted membership in the organization. When `false`, users must be " +
+						"granted membership in the organization before logging in with this connection.",
+				},
+			},
+		},
+	}
+
+	dataSourceSchema["members"] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"user_id": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The ID of the organization member.",
+				},
+				"name": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The name of the organization member.",
+				},
+				"email": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The email of the organization member.",
+				},
+			},
+		},
+	}
+
+	return dataSourceSchema
+}
+
+func readOrganizationForDataSource(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+	var foundOrganization *management.Organization
+	var err error
+
+	organizationID := data.Get("organization_id").(string)
+	if organizationID != "" {
+		foundOrganization, err = api.Organization.Read(organizationID)
+		if err != nil {
+			if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+				data.SetId("")
+				return nil
+			}
+			return diag.FromErr(err)
+		}
+	} else {
+		name := data.Get("name").(string)
+		page := 0
+
+	outerLoop:
+		for {
+			organizations, err := api.Organization.List(management.Page(page))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			for _, organization := range organizations.Organizations {
+				if organization.GetName() == name {
+					foundOrganization = organization
+					break outerLoop
+				}
+			}
+
+			if !organizations.HasNext() {
+				break
+			}
+
+			page++
+		}
+
+		if foundOrganization == nil {
+			return diag.Errorf("No organization found with \"name\" = %q", name)
+		}
+	}
+
+	data.SetId(foundOrganization.GetID())
+
+	result := multierror.Append(
+		data.Set("name", foundOrganization.GetName()),
+		data.Set("display_name", foundOrganization.GetDisplayName()),
+		data.Set("branding", flattenOrganizationBranding(foundOrganization.GetBranding())),
+		data.Set("metadata", foundOrganization.GetMetadata()),
+	)
+
+	foundConnections, err := fetchAllOrganizationConnections(api, foundOrganization.GetID())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	result = multierror.Append(
+		result,
+		data.Set("connections", flattenOrganizationConnections(foundConnections)),
+	)
+
+	foundMembers, err := fetchAllOrganizationMembers(api, foundOrganization.GetID())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	result = multierror.Append(
+		result,
+		data.Set("members", flattenOrganizationMembers(foundMembers)),
+	)
```

Comment:
Any performance concerns about retrieving all connections and members? Connections _should_ be a reasonable quality but it is possible for an organization to possess a large number of members.

Reply:
I believe there's a limit on the API, they only return 1000 items at most.